File: clojure/gigasecond/src/gigasecond.clj
(ns gigasecond
  (:import java.util.Calendar))

(defn- format [date]
  [(.get date Calendar/YEAR) (inc (.get date Calendar/MONTH)) (.get date Calendar/DATE)])

(defn from [year month day]
  (format (doto (Calendar/getInstance)
    (.set year (dec month) day)
    (.add Calendar/SECOND 1000000000))))

File: scala/gigasecond/src/main/scala/Gigasecond.scala
import java.time.{LocalDate, LocalDateTime}

object Gigasecond {
  def addGigaseconds(date: LocalDate): LocalDateTime = addGigaseconds(date.atStartOfDay)
  def addGigaseconds(datetime: LocalDateTime): LocalDateTime = datetime.plusSeconds(1000000000)
}
